pilot/pkg/security/authz/builder/v2.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package builder
+
+import (
+	"fmt"
+	"strings"
+
+	http_config "github.com/envoyproxy/go-control-plane/envoy/config/filter/http/rbac/v2"
+	envoy_rbac "github.com/envoyproxy/go-control-plane/envoy/config/rbac/v2"
+
+	istio_rbac "istio.io/api/rbac/v1alpha1"
+	"istio.io/istio/pilot/pkg/model"
+)
+
+const (
+	// rootNamespacePrefix is the prefix of the root namespace. This is used to refer to ServiceRoles
+	// that are defined in the root namespace.
+	// For more details, check out root_namespace field at https://github.com/istio/api/blob/master/mesh/v1alpha1/config.proto
+	rootNamespacePrefix = "/"
+)
+
+// buildV2 builds the v2 policy to filter config.
+func (b *Builder) buildV2(forTCPFilter bool) *http_config.RBAC {
```

Comment:
for tcp filter, why return a http rbac proto ?

Reply:
forTCPFilter is whether the config is for TCP or HTTP service. I think RBAC filter is part of the Envoy `HttpConnectionManager`, that's why we're returning http config for both cases, but I could be wrong.

Reply:
Previously we only support HTTP filter and when adding support for TCP filter, we simply reused the previous code and extract the actual RBAC config from `http_config.RBAC` and put it in the Envoy TCP filter config. 
We could further refactor this to make it return the real RBAC configs instead of wrapping it in the HTTP filter config.